src/user/mod.rs

```diff
@@ -378,6 +378,24 @@ pub fn updates(req: &mut Request) -> CargoResult<Response> {
     Ok(req.json(&R{ versions: versions, meta: Meta { more: more } }))
 }
 
+/// Handles the `GET /users/:user_id/stats` route.
+pub fn stats(req: &mut Request) -> CargoResult<Response> {
+    use  diesel::expression::dsl::sum;
+    let name = &req.params()["user_id"].parse::<i32>().ok().unwrap();
+    let conn = req.db_conn()?;
+    
+    let data = crate_downloads::table.filter(crate_downloads::crate_id.eq_any(
+        crate_owners::table.select(crate_owners::crate_id)
+            .filter(crate_owners::owner_id.eq(name))
+        )).select(sum(crate_downloads::downloads)).first::<i64>(&*conn)?; 
```

Comment:
> Since we're sending user_id from the dashboard, we'll be basically querying crate_owners with owner_kind as OwnerKind::User only right? even if it's not explicitly laid out. Do you think to convey the intent properly we should add this as a filter as well?

The id is *never* enough to specify that it's a user on its own. Users and teams can have the same ID, as they live on different tables.

> Is there any other approach I can take to make inner_join work? Thanks! :)

I was going to link to the macro, but it's not rendering in our docs for some reason. You can use the `joinable!` macro for this: `joinable!(child_table -> parent_table (foreign_key_name))`